addon/components/pagination.hbs

```diff
@@ -21,15 +21,95 @@
           {{on "click" this.previous}}
         >{{@previousText}}</a>
       </li>
-      {{#each this.pages as |_ index|}}
-        <li class="page-item {{if (eq @page index) 'active'}}">
-          <a
-            class="page-link"
-            href="#"
-            {{on "click" (fn this.change index)}}
-          >{{inc index}}</a>
-        </li>
-      {{/each}}
+      {{#if (lt this.length 6)}}
+        {{#each this.pages as |_ index|}}
+          <li class="page-item {{if (eq @page index) 'active'}}">
```

Comment:
Can you extract this duplicate code into a new component called pagination/item and then use that inside each one of these conditionals? No need to add it to the template-registry or index bc it will only be used internally here, to reduce duplication.